internal/entities/quote/pegin_quote.go

```diff
@@ -30,14 +29,22 @@ type PeginQuoteRepository interface {
 	InsertQuote(ctx context.Context, quote CreatedPeginQuote) error
 	GetQuote(ctx context.Context, hash string) (*PeginQuote, error)
 	GetPeginCreationData(ctx context.Context, hash string) PeginCreationData
+	GetQuotes(ctx context.Context, hashes []string) ([]PeginQuote, error)
 	GetRetainedQuote(ctx context.Context, hash string) (*RetainedPeginQuote, error)
 	InsertRetainedQuote(ctx context.Context, quote RetainedPeginQuote) error
 	UpdateRetainedQuote(ctx context.Context, quote RetainedPeginQuote) error
 	GetRetainedQuoteByState(ctx context.Context, states ...PeginState) ([]RetainedPeginQuote, error)
+	GetQuotesByState(ctx context.Context, filter GetPeginQuotesByStateFilter) ([]PeginQuote, error)
 	// DeleteQuotes deletes both regular and retained quotes
 	DeleteQuotes(ctx context.Context, quotes []string) (uint, error)
 }
 
+type GetPeginQuotesByStateFilter struct {
```

Comment:
same comment as for the function, the name of this doesn't reflect the reality its a filter by state but also has dates inside